### DISCUSSION TITLE
Making compiler

### DISCUSSION THREAD

**Discussion by DEVdv2006** [community/community#51768] [Programming Help]
### Body

Hi,i am currently CE student and I want to devlop a compiler that can compile multiple languages at same time.for example ,some part of code written in Java rest in python so it can compile and execute it properly , is it possible, if so what skill sets are required 

### I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

- [X] Yes

**Comment by mpboom** [comment]
Writing a compiler is a very difficult task that involves complex concepts relating to different stages and technologies parsers use. I'd recommend starting a bit smaller than building some multi-language compiler, but rather start off with small examples. Think about what your target language is (Assembly, or - maybe a bit easier - another higher level language). I also see some issues with your plans of reconicling multiple programming languages into one parser as different languages sometimes have conflicting properties which would make that impossible, extremely difficult or unperformant.

Try finding information about parsing in general. What are the stages of tokenizing, lexing and parsing. How can you mathematically describe a "language" and what is an AST? Find articles or books online, or ask (specific) questions if you get stuck. Good luck.

**Reply by DEVdv2006** [reply]
Thanks for your valuable guidance

**Comment by sergeyleschev** [comment]
Hi DEVdv2006, developing a compiler that can compile multiple languages at the same time can be a challenging task. You'll need a strong understanding of programming concepts and language syntax to accomplish this. Some recommended skill sets are expertise in programming languages such as Java or C++, knowledge of compilers' theory and implementation, and experience with parsing and code generation techniques. Additionally, you may want to consider collaborating with other developers who have specialized expertise in the different languages you want to support. Good luck with your project!
